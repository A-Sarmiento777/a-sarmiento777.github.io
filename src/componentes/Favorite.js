import React from 'react';
import { useHistory } from 'react-router';

const Favorite = ({ carritoFav, agregarProductoAlCarrito, removeItemFromCartFav, clearFav }) => {
    const history = useHistory()
    let amount = carritoFav?.map(a => a.price)
    let totalAmount = amount?.reduce((a, b) => a + b, 0)
    const addFavToCart = async () => {
        await carritoFav?.map((producto) => agregarProductoAlCarrito(producto?.id, producto?.name, producto?.price, producto?.type, 1))
        clearFav()
        history.push('/checkout')
    }
    return (
        <div className="card p-4 mt-5 pt-4 mb-5 rounded" style={{ minWidth: "460px" }}>
            {carritoFav.length ? <table style={{ width: '100%' }}>
                <thead>
                    <tr>
                        <td style={{ fontWeight: 'bold' }}>Product</td>
                        <td style={{ fontWeight: 'bold' }}>Title</td>
                        <td style={{ fontWeight: 'bold' }}>Price</td>
                        <td style={{ fontWeight: 'bold' }}>Remove</td>
                    </tr>
                </thead>
                <tbody>
                    {carritoFav?.map((producto, index) => {
                        return (
                            <>
                                <tr key="index">
                                    <td><img src={process.env.PUBLIC_URL + `/Assets/${producto.type}/${producto.name}.jpg`} width='100' alt="logos" className="img-fluid py-2" /></td>
                                    <td>{producto.name}</td>
                                    <td>{producto.price}</td>
                                    <td onClick={() => removeItemFromCartFav(producto.id)} style={{ color: 'red' }}>Remove</td>
                                </tr>
                            </>
                        )
                    })}
                </tbody>
            </table> : <h4>Nothing in Fav! Please Add some items</h4>}

            {carritoFav.length ?
                <div className="mx-auto">
                    <div>
                        Price: {totalAmount - ((totalAmount * 25) / 100) + ' $'}

                    </div>
                    <div>
                        Incl tax: {totalAmount + ' $'}

                    </div>
                    <div className="text-center mx-auto">
                        <button className="p-1 rounded" style={{ backgroundColor: 'black', color: 'white' }} onClick={addFavToCart}>Add to Cart from Favourites</button>
                        {/* <button className="p-1 rounded" style={{ backgroundColor: 'black', color: 'white' }} onClick={() => {
                        clearFav()
                        history.push('/home')
                    }}>Clear Favourites</button> */}
                    </div>
                </div>
                : null}
        </div >
    );
}


export default Favorite;