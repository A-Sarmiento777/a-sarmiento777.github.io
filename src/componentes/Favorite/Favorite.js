import React from 'react';
import { useHistory } from 'react-router';
import FavoriteCard from '../FavoriteCard/FavoriteCard';

const Favorite = ({ carritoFav, agregarProductoAlCarrito, cart, removeItemFromCartFav, clearFav }) => {
    const history = useHistory()
    let amount = carritoFav?.map(a => a.price)
    let totalAmount = amount?.reduce((a, b) => a + b, 0)

    const addFavToCart = async () => {
        let data = carritoFav.filter(item1 => {
            let index = cart.findIndex(item2 => item1.id === item2.id);
            if(index < 0) return true
            
            return false;
            });
        await data?.map((producto) => agregarProductoAlCarrito(producto?.id, producto?.name, producto?.price, producto?.type, 1))
        clearFav()
        history.push('/checkout')
    }
    return (
        <div className="card p-4 mt-5 pt-4 mb-5 rounded" style={{ minWidth: "460px" }}>
            {carritoFav.length ? <table style={{ width: '100%' }}>
                <thead>
                    <tr>
                        <td style={{ fontWeight: 'bold' }}>Product</td>
                        <td style={{ fontWeight: 'bold' }}>Title</td>
                        <td style={{ fontWeight: 'bold' }}>Price</td>
                        <td style={{ fontWeight: 'bold' }}>Remove</td>
                    </tr>
                </thead>
                <tbody>
                    {carritoFav?.map((producto, index) => {
                        return (
                            <FavoriteCard key={index + 1} removeItemFromCartFav={removeItemFromCartFav} producto={producto} />
                        )
                    })}
                </tbody>
            </table> : <h4>Nothing in Fav! Please Add some items</h4>}

            {carritoFav.length ?
                <div className="mx-auto">
                    <div>
                        Price: {totalAmount - ((totalAmount * 25) / 100) + ' kr'}

                    </div>
                    <div>
                        Incl moms: {totalAmount + ' kr'}

                    </div>
                    <div className="text-center mx-auto">
                        {/* <button className="p-1 rounded" style={{ backgroundColor: 'black', color: 'white' }} onClick={() => {
                        clearFav()
                        history.push('/home')
                    }}>Clear Favourites</button> */}
                        <button className="p-1 rounded" style={{ backgroundColor: 'black', color: 'white' }} onClick={addFavToCart}>Add to Cart from Favourites</button>
                    </div>
                </div>
                : null}

        </div >
    );
}


export default Favorite;