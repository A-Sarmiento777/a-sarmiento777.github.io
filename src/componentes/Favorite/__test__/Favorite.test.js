import React from "react";
import { act, create, waitFor } from "react-test-renderer";
import Favorite from "../Favorite";

test("It should render completed Favorite", async () => {
  jest.spyOn(global, "fetch").mockImplementation(() => {
    return Promise.resolve({
      json: () => Promise.resolve([]),
    });
  });

  let testRenderer;
  await act(async () => {
    testRenderer = create(
        <Favorite carritoFav={[{ id: 1, type: 'saadad', name: "Test", price: "6.7" }]} removeItemFromCartFav={() => {}} clearFav={() => {}}/>
        );
  });

  let tree = testRenderer.toJSON();
  expect(tree).toMatchSnapshot();
//   expect(global.fetch).toHaveBeenCalledTimes(1);
});
