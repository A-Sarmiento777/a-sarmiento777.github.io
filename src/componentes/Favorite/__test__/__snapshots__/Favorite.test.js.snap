// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should render completed Favorite 1`] = `
<div
  className="card p-4 mt-5 pt-4 mb-5 rounded"
  style={
    Object {
      "minWidth": "460px",
    }
  }
>
  <table
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <thead>
      <tr>
        <td
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          Product
        </td>
        <td
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          Title
        </td>
        <td
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          Price
        </td>
        <td
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          Remove
        </td>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          <img
            alt="logos"
            className="img-fluid py-2"
            src="/Assets/saadad/Test.jpg"
            width="100"
          />
        </td>
        <td>
          Test
        </td>
        <td />
        <td
          onClick={[Function]}
          style={
            Object {
              "color": "red",
            }
          }
        >
          Remove
        </td>
      </tr>
    </tbody>
  </table>
  <div
    className="mx-auto"
  >
    <div>
      Price: 
      undefined $
    </div>
    <div>
      Incl tax: 
      undefined $
    </div>
    <div
      className="text-center mx-auto"
    >
      <button
        className="p-1 rounded"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "black",
            "color": "white",
          }
        }
      >
        Add to Cart from Favourites
      </button>
    </div>
  </div>
</div>
`;
